flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/OperatorChain.java

```diff
@@ -591,16 +591,18 @@ public void collect(StreamRecord<T> record) {
 				operator.setKeyContextElement1(copy);
 				operator.processElement(copy);
 			} catch (ClassCastException e) {
-				// Enrich error message
-				ClassCastException replace = new ClassCastException(
-					String.format(
-						"%s. Failed to push OutputTag with id '%s' to operator. " +
-						"This can occur when multiple OutputTags with different types " +
-						"but identical names are being used.",
-						e.getMessage(),
-						outputTag.getId()));
-
-				throw new ExceptionInChainedOperatorException(replace);
+				if (outputTag != null) {
```

Comment:
you are now completely swallowing the exception if the outputtag is null which is unacceptable.

Reply:
Okay. So, should we try catch NPE or just output the null id ?

Reply:
Neither. just add an else block that re-throws the original exception.

Reply:
Thanks @zentol A good method!